api/client/client.go

```diff
@@ -3608,13 +3608,78 @@ func ListUnifiedResourcePage(ctx context.Context, clt ListUnifiedResourcesClient
 	}
 }
 
+// GetEnrichedResourcePage is a helper for getting a single page of enriched resources.
+func GetEnrichedResourcePage(ctx context.Context, clt GetResourcesClient, req *proto.ListResourcesRequest) (ResourcePage[*types.EnrichedResource], error) {
+	var out ResourcePage[*types.EnrichedResource]
+
+	// Set the limit to the default size if one was not provided within
+	// an acceptable range.
+	if req.Limit <= 0 || req.Limit > int32(defaults.DefaultChunkSize) {
+		req.Limit = int32(defaults.DefaultChunkSize)
+	}
```

Comment:
does not require a change because this is how we are currently doing it:

Setting the limit to default if not supplied is great. 
Overwriting if it's greater tho might be confusing if I'm building a caller that receives "successful" responses with too large of a limit. It might be better to return an error and say "hey thats too much" so I know that my request is shaped wrong.

Again, don't need to change here. I'll inspect how other APIs handle this case